File: programmers/level1/핸드폰 번호 가리기/solution.cpp
#include <string>
#include <vector>

using namespace std;

string createSecretPhoneNumber(string& phoneNumber);

string solution(string phoneNumber) {
    return createSecretPhoneNumber(phoneNumber);
}

string createSecretPhoneNumber(string& phoneNumber){
    for (int i = 0; i < phoneNumber.size() - 4; ++i)
        phoneNumber[i] = '*';
    return phoneNumber;
}

File: programmers/level1/핸드폰 번호 가리기/solution.js
const DELIMITER = "";
const BACK_NUMBER_LENGTH = 4;

function solution(phoneNumber) {
  return createHidedPhoneNumber(phoneNumber);
}

const createHidedPhoneNumber = (phoneNumber) => {
  return phoneNumber
    .split(DELIMITER)
    .map(hideNumber(phoneNumber.length - BACK_NUMBER_LENGTH))
    .join(DELIMITER);
};

const hideNumber = (length) => (char, index) => index < length ? "*" : char;
